tests/test_conversion_cyclical.py

```diff
@@ -0,0 +1,149 @@
+# Imports sorted alphabetically
+import pathlib
+import pytest
+
+# 'from' imports, sorted alphabetically by module
+from google.protobuf.json_format import Parse
+from google.protobuf.message import Message
+from protarrow.common import M, ProtarrowConfig
+from protarrow.proto_to_arrow import (
+    messages_to_record_batch,
+    ProtarrowCycleError,
+    messages_to_table,
+)
+from protarrow_protos.bench_pb2 import (
+    CyclicalDirectMessage,
+    CyclicalIndirectMessageLevel1,
+    CyclicalMapMessage,
+    CyclicalRepeatedMessage,
+)
+from typing import List, Type
+
+CONFIGS = [
+    ProtarrowConfig(purge_cyclical_messages=False),
+    ProtarrowConfig(purge_cyclical_messages=True),
+]
+DIR = pathlib.Path(__file__).parent
+
+
+def read_proto_jsonl(path: pathlib.Path, message_type: Type[M]) -> List[M]:
+    with path.open() as fp:
+        return [
+            Parse(line.strip(), message_type())
+            for line in fp
+            if line.strip() and not line.startswith("#")
+        ]
+
+
+def _load_data(filename: str, message_type: Type[Message]) -> List[Message]:
```

Comment:
Some of this code you can import from the other tests. Or move it to `tests/data_utils.py`